app/classifier/tasks/highlighter/label-renderer.jsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+
+const cache = {}; 
+
+export default class LabelRenderer extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.createNewContent = this.createNewContent.bind(this);
+    this.state = {
+      content: ''
+    }
+  }
+
+  componentWillReceiveProps(newProps) {
+    if (newProps.src !== this.props.src) {
+      this.loadText(newProps.src);
+    }
+  }
+
+  componentDidMount(){
+    this.loadText(this.props.src);
+  }
+
+  loadText(src) {
```

Comment:
This is duplicated from the `TextViewer` component. It would be good to get rid of it and rely on the subject viewer to load our subject for us.